### ISSUE TITLE
Improve Responsiveness of the UI when loading

### ISSUE THREAD

**Issue by sitefinitysteve** [Sitefinity/feather#1716]
Right now if I clear cache and drop on a content block widget, wait till it gives the "Create Content" icon...click THAT...there's a good 5-10 seconds of waiting before anything visibly happens.  Now personally know something is happening, and I can see scripts and things coming down in the dev tools, but out users don't seem to know this and it's resulting in them clicking the edit buttons multiple times...

Likewise in the video it takes forever for the images to appear in the UI giving the appearance that we have no images

**Comment by ElenaGaneva** [comment]
Hi Steve,

Thank you for bringing this to our attention. We already have fix for this on pages based on Bootstrap, Foundation and SemanticUI layouts, still it is reproducible on hybrid. Therefore I am adding this to the backlog, so we can investigate it further.

Thank you for your feedback.

Regards,
Elena

**Comment by sitefinitysteve** [comment]
Yeah you're right, in Pure it's okay, hybrid not so much

However the load time even in pure is crazy at like 20 seconds?  http://screencast.com/t/irRWE6UaCQn  ...can that be improved at all?

**Comment by sitefinitysteve** [comment]
@mariushm Any movement on this?  ...our editors hate using the MVC widgets right now because of the loading lag...but they're SO MUCH BETTER, taints them a bit

**Comment by mariushm** [comment]
This won't make it for the 8.1 release, but we will do it shortly after. 

Do you experience this on second load of the widget or designer?

**Comment by sitefinitysteve** [comment]
Nope, once they're cached it's great, but initally they click "edit" and nothing happens for a while (so they mash edit), or they see the loading icon for a while and get frustrated because "it's slow".

**Comment by mariushm** [comment]
I see, it definitely ruins the first impression. We will take care of that shortly.

**Comment by sitefinitysteve** [comment]
Appreciate it, thanks so much

**Comment by sitefinitysteve** [comment]
Any movement on this?  The issue of the overlay seems to be resolved now, but the "loading" time is still pretty obnoxious.
